File: remove_bg.php
<?php
$files = glob("assets/icons/*.png");

foreach ($files as $file) {
    echo "Processing $file...\n";
    $img = imagecreatefrompng($file);
    if (!$img) {
        echo "Failed to load $file\n";
        continue;
    }
    
    // Enable alpha blending
    imagealphablending($img, false);
    imagesavealpha($img, true);
    
    $width = imagesx($img);
    $height = imagesy($img);
    
    // Target background color: #0f172a (15, 23, 42)
    $targetR = 15;
    $targetG = 23;
    $targetB = 42;
    $tolerance = 60; // How far off the color can be
    
    for ($x = 0; $x < $width; $x++) {
        for ($y = 0; $y < $height; $y++) {
            $color = imagecolorat($img, $x, $y);
            $r = ($color >> 16) & 0xFF;
            $g = ($color >> 8) & 0xFF;
            $b = $color & 0xFF;
            
            // Calculate distance
            $dist = sqrt(pow($r - $targetR, 2) + pow($g - $targetG, 2) + pow($b - $targetB, 2));
            
            if ($dist < $tolerance) {
                // Closer to background -> more transparent
                $alpha = 127; 
                // Feathering:
                if ($dist > ($tolerance - 20)) {
                    $alpha = (int)(127 * (($tolerance - $dist) / 20));
                    $alpha = 127 - $alpha; // 127 is fully transparent, 0 is fully opaque
                }
                
                $newColor = imagecolorallocatealpha($img, $r, $g, $b, $alpha);
                imagesetpixel($img, $x, $y, $newColor);
            }
        }
    }
    
    imagepng($img, $file);
    imagedestroy($img);
    echo "Saved $file\n";
}

File: remove_bg.py
import os
import glob
from PIL import Image
import math

def process_image(path):
    print(f"Processing {path}")
    img = Image.open(path).convert("RGBA")
    data = img.getdata()

    new_data = []
    # Target color
    tr, tg, tb = 15, 23, 42
    tolerance = 80

    for item in data:
        r, g, b, a = item
        # Euclidean distance
        dist = math.sqrt((r - tr)**2 + (g - tg)**2 + (b - tb)**2)

        if dist < tolerance:
            # Fully transparent if very close, feathering if further away
            if dist < tolerance - 30:
                new_data.append((r, g, b, 0))
            else:
                alpha = int(((dist - (tolerance - 30)) / 30) * 255)
                new_data.append((r, g, b, alpha))
        else:
            new_data.append((r, g, b, a))

    img.putdata(new_data)
    img.save(path, "PNG")

files = glob.glob("assets/icons/*.png")
for f in files:
    process_image(f)

print("Done")
